advisor/src/main/kotlin/Advisor.kt

```diff
@@ -56,7 +57,7 @@ class Advisor(
      * can optionally be [skipped][skipExcluded].
      */
     @JvmOverloads
-    fun advise(ortResult: OrtResult, skipExcluded: Boolean = false): OrtResult {
+    suspend fun advise(ortResult: OrtResult, skipExcluded: Boolean = false): OrtResult {
```

Comment:
Does this change make it harder to call the API from Java? Mind giving a hint why it becomes easier to intergate? (just asking to learn as I'm not yet deep into Kotlin's coroutines).

Reply:
> Does this change make it harder to call the API from Java?

I was wondering about this as well, and it [seems like](https://stackoverflow.com/a/71599237/1127485) you'd need a `CompletableFuture` wrapper in that case. Put short, `suspend`functions are not directly callable from Java, so if we're fine with what, you could drop `@JvmOverloads` from here as well as a consequence.

Reply:
More reading on that topic: https://jorgecastillo.dev/digging-into-kotlin-continuations

Reply:
@fviernau, one big advantage of coroutines is that you can do things concurrently with a low number of threads. In this model, an action that would traditionally block a thread is put aside, and the thread can continue to run and process other actions available. You need to be careful, however, when calling code that is not prepared for this model and that might block in the traditional way; this would potentially impact all your coroutines. Therefore, you would typically take special care to prevent this, e.g. by having a special executor with a dedicated threadpool for running such blocking code.

By marking the functions of the Advisor with the *suspend* keyword, the component explicitly indicates that it adheres to the coroutine programming model and can be used directly without additional means.

Reply:
I still believe `@JvmOverloads` should be dropped.

Reply:
Done.